test/server.test.js

```diff
@@ -1076,6 +1076,34 @@ test('gh-283 maximum available versioned route matching', function (t) {
     });
 });
 
+test('routes matching multiple versions should know what the max matched version was', function (t) {
```

Comment:
please add `gh #635` to the beginning of your test case, and this should be good to go!